From 41c28a602edf24d2d1a933064a83b8b6fd189c29 Mon Sep 17 00:00:00 2001
From: Mel Gorman <mgorman@suse.de>
Date: Mon, 10 Mar 2014 13:30:44 +0000
Subject: [PATCH] Revert "PM / reboot: call syscore_shutdown() after
 disable_nonboot_cpus()"

References: Reduce time to shutdown large machines (bnc#865442)
Patch-mainline: No (Unnecessary)

This reverts commit 44a44be00a8a547855c4265d96b1a42261f26e8a. The patches
were only required for the ARM architecture which we do not support but
a consequence of the patch is that large machines take too long to shutdown.
Mainline addresses the problem differently but the patches are unnecessarily
complex for backporting.

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 kernel/sys.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/kernel/sys.c b/kernel/sys.c
index 10a2161..ddd9060 100644
--- a/kernel/sys.c
+++ b/kernel/sys.c
@@ -318,6 +318,7 @@ void kernel_restart_prepare(char *cmd)
 	system_state = SYSTEM_RESTART;
 	usermodehelper_disable();
 	device_shutdown();
+	syscore_shutdown();
 }
 
 /**
@@ -332,7 +333,6 @@ void kernel_restart(char *cmd)
 {
 	kernel_restart_prepare(cmd);
 	disable_nonboot_cpus();
-	syscore_shutdown();
 	if (!cmd)
 		printk(KERN_EMERG "Restarting system.\n");
 	else
@@ -358,7 +358,6 @@ static void kernel_shutdown_prepare(enum system_states state)
 void kernel_halt(void)
 {
 	kernel_shutdown_prepare(SYSTEM_HALT);
-	disable_nonboot_cpus();
 	syscore_shutdown();
 	printk(KERN_EMERG "System halted.\n");
 	kmsg_dump(KMSG_DUMP_HALT);
